lib/rubocop/cop/rspec/aggregate_examples.rb

```diff
@@ -0,0 +1,374 @@
+      #    of the `not_to` expectations is barely possible when a matcher
+      #    doesn't provide a negated variant.
+      #
+      # 3. Aggregation of block syntax with non-block syntax should be in a
+      #    specific order.
+      #
+      #
+      # The examples containing matchers that leave subject in modified state
+      # will fail when not supposed to or come with a risk of not failing when
+      # expected to fail if aggregated. One example is
+      # `validate_presence_of :comment` as it leaves an empty comment after
+      # itself on the subject making it invalid and the subsequent expectation
+      # to fail. Examples with those matchers are not supposed to be aggregated.
+      #
+      # @example MatchersWithSideEffects
+      #
+      #   # .rubocop.yml
+      #   # RSpec/AggregateExamples:
+      #   #   MatchersWithSideEffects:
+      #   #   - allow_value
+      #   #   - allow_values
+      #   #   - validate_presence_of
+      #
+      #   # bad, but is not automatically correctable
+      #   describe do
+      #     it { is_expected.to validate_presence_of(:comment) }
+      #     it { is_expected.to be_valid }
+      #   end
+      #
+      #
+      # RSpec [comes with an `aggregate_failures` helper](https://relishapp.com/rspec/rspec-expectations/docs/aggregating-failures)
+      # not to fail the example on first unmet expectation that might come
+      # handy with aggregated examples.
+      # It can be [used in metadata form](https://relishapp.com/rspec/rspec-core/docs/expectation-framework-integration/aggregating-failures#use-%60:aggregate-failures%60-metadata),
+      # or [enabled globally](https://relishapp.com/rspec/rspec-core/docs/expectation-framework-integration/aggregating-failures#enable-failure-aggregation-globally-using-%60define-derived-metadata%60).
+      #
+      # @example Globally enable `aggregate_failures`
+      #
+      #   # spec/spec_helper.rb
+      #   config.define_derived_metadata do |metadata|
+      #     unless metadata.key?(:aggregate_failures)
+      #       metadata[:aggregate_failures] = true
+      #     end
+      #   end
+      #
+      # To match the style being used in the spec suite, AggregateExamples
+      # can be configured to add metadata to the example or not. The option
+      # not to add metadata can be also used when it's not desired to make
+      # expectations after previously failed ones commonly known as fail-fast.
+      #
+      # @example AddAggregateFailuresMetadata: false (default)
+      #
+      #   specify do
+      #     expect(number).to be_positive
+      #     expect(number).to be_odd
+      #   end
+      #
+      # @example AddAggregateFailuresMetadata: true
+      #
+      #   specify(:aggregate_failures) do
+      #     expect(number).to be_positive
+      #     expect(number).to be_odd
+      #   end
+      #
+      class AggregateExamples < Cop # rubocop:disable Metrics/ClassLength
```

Comment:
It would be nice to (perhaps in a later PR) enable `Metrics/ClassLength` on this class.

Reply:
Actually (as mentioned earlier) I’m having a really hard time reviewing this class. Which suggests that it will be hard to maintain…

Would it be possible to extract some logic into separate classes (just keep the in the same time, namespaced within `AggregateExamples` would be fine. I’d suggest grouping some of the autocorrect code (`range_for_replace`, `aggregated_example`, `drop_example`) in a class, or the “metadata” methods (`metadata_without_aggregate_failures`, `metadata_symbols_without_aggregate_failures`, `metadata_pairs_without_aggegate_failures`, `aggregate_failures_disabled`)?

Reply:
Now it's been almost a month since I last touched this cop, and I feel your pain when I look at the code.
Will structure to it, and will add clarifications to the code that is not obvious.
Thanks for the suggestion.

Reply:
>  (just keep the in the same time, namespaced within `AggregateExamples` would be fine.

That’s the kind of language that comes from doing code review on my phone… 😆 I think I meant to say “the new classes can just be defined inside the `AggregateExamples` namespace.”